docs/core/extensions/caching.md

```diff
@@ -0,0 +1,179 @@
+
+If the `cache` contains the `letter` key, and the `value` is an instance of an `AlphabetLetter` it's written to the console. When the `letter` key is not in the cache, it was evicted and its post eviction callback was invoked.
+
+#### Put it all together
+
+The entire sample app source code is a top-level program and requires two NuGet packages:
+
+- [`Microsoft.Extensions.Caching.Memory`](/dotnet/api/microsoft.extensions.caching.memory)
+- [`Microsoft.Extensions.Hosting`](/dotnet/api/microsoft.extensions.hosting)
+
+:::code source="snippets/caching/memory-apis/Program.cs":::
+
+Feel free to adjust the `MillisecondsDelayAfterAdd` and `MillisecondsAbsoluteExpiration` values to observe the changes in behavior to the expiration and eviction of cached entries. The following is sample output from running this code, due to the non-deterministic nature of .NET events &mdash; there is no guarantee that your output will be identical.
+
+```console
+A was cached.
+B was cached.
+C was cached.
+D was cached.
+E was cached.
+F was cached.
+G was cached.
+H was cached.
+I was cached.
+J was cached.
+K was cached.
+L was cached.
+M was cached.
+N was cached.
+O was cached.
+P was cached.
+Q was cached.
+R was cached.
+S was cached.
+T was cached.
+U was cached.
+V was cached.
+W was cached.
+X was cached.
+Y was cached.
+Z was cached.
+
+Q is still in cache. The 'Q' character is the 17 letter in the English alphabet.
+R is still in cache. The 'R' character is the 18 letter in the English alphabet.
+S is still in cache. The 'S' character is the 19 letter in the English alphabet.
+T is still in cache. The 'T' character is the 20 letter in the English alphabet.
+U is still in cache. The 'U' character is the 21 letter in the English alphabet.
+D was evicted for Expired.
+C was evicted for Expired.
+G was evicted for Expired.
+E was evicted for Expired.
+F was evicted for Expired.
+B was evicted for Expired.
+M was evicted for Expired.
+V is still in cache. The 'V' character is the 22 letter in the English alphabet.
+H was evicted for Expired.
+I was evicted for Expired.
+J was evicted for Expired.
+K was evicted for Expired.
+L was evicted for Expired.
+A was evicted for Expired.
+N was evicted for Expired.
+W is still in cache. The 'W' character is the 23 letter in the English alphabet.
+O was evicted for Expired.
+P was evicted for Expired.
+X is still in cache. The 'X' character is the 24 letter in the English alphabet.
+Y is still in cache. The 'Y' character is the 25 letter in the English alphabet.
+Z is still in cache. The 'Z' character is the 26 letter in the English alphabet.
+```
+
+Since the absolute expiration is set, all the cached items will eventually be evicted.
+
+## Distributed caching
+
+In some scenarios, a distributed cache is required. A distributed cache can support higher scale-out than an in-memory cache. Using a distributed cache offloads the cache memory to an external process.
+
+<!-- TODO: Add a lot more content here. -->
```

Comment:
I presume you were going to talk about things like Redis and Azure Cache for Redis?

Reply:
That's the plan, yes.